documentation/tig_stack.md

```diff
@@ -0,0 +1,340 @@
+# Deploy a TIG stack with Bolt
+
+In this guide, you'll use Bolt to configure and deploy a TIG stack. 
+
+A TIG stack provides metrics visualization to help you monitor your
+infrastructure. TIG stands for:
+- [Telegraf](https://docs.influxdata.com/telegraf/v1.14/) - A plugin-driven
+  server agent for collecting and reporting metrics.
+- [InfluxDB](https://docs.influxdata.com/influxdb/v1.8/) - A time series
+  database designed to handle time-stamped data like metrics and events.     
+- [Grafana](https://grafana.com/docs/grafana/latest/getting-started/what-is-grafana/) -
+  An open source visualization and analytics tool.
+
+> **Before you begin**
+>
+> - Make sure you've installed Bolt version 1.12.0 or greater on your machine.
+>  For instructions on how to install Bolt, see
+>  [Installing Bolt](./bolt_installing.md).
+> - Make sure you've installed Vagrant and a hypervisor like 
+>  [VirtualBox](https://www.virtualbox.org/). This guide uses virtual machines
+>  to host the stack. If you'd prefer to use your own hosts, you can skip this
+>  step.
+> - Clone or download the [TIG stack repo](https://github.com/puppetlabs/bolt-tig-stack)
+
+To deploy your TIG stack, you'll use a Bolt plan that leverages existing Puppet
+Modules. You can find all the files you need in the
+[TIG stack repo](https://github.com/puppetlabs/bolt-tig-stack).
+
+After you've cloned or downloaded the repo, you can access the TIG Bolt
+project from the `tig_stack` directory. The project has the following file structure:
+
+```shell
+.
+├── Puppetfile
+├── Vagrantfile
+├── bolt-project.yaml
+├── hiera.yaml
+├── inventory.yaml
+├── manifests
+│   ├── dashboard.pp
+│   ├── params.pp
+│   └── telegraf.pp
+├── plans
+│   └── init.pp
+└── templates
+    └── dashboards
+        └── telegraf.json
+```
+
+## Provision your targets
+
+A typical real-world environment would consist of multiple agents, but for the
+purposes of this guide, you'll provision two virtual machines to use as
+targets - `target0` and `target1`. The first target hosts the Garafana dashboard
+and the InfluxDB database as well as a Telegraf agent. The second target hosts
+only a Telegraf agent. If you'd prefer to use your own existing targets, skip
+this section.
+
+The `bolt_tig` directory contains the following `Vagrantfile`:
+
+```shell
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+TARGETS = 2
+
+Vagrant.configure(TARGETS) do |config|
+  config.vm.box = "centos/7"
+  config.ssh.forward_agent = true
+
+  TARGETS.times do |i|
+    config.vm.define "target#{i}" do |target|
+      target.vm.hostname = "target#{i}"
+      target.vm.network :private_network, ip: "10.0.0.#{100 + i}"
+    end
+  end
+end
+```
+
+To provision your targets:
+1. Generate the SSH configuration for both targets. Bolt will automatically
```

Comment:
First step should be running `vagrant up` from the project directory.

It might also be worth putting a note here that vagrant may not be able to configure the private network if it collides with an existing one (for example the wifi network). An example of the error it will raise:

```shell
The specified host network collides with a non-hostonly network!
This will cause your specified IP to be inaccessible. Please change
the IP or name of your host only network so that it no longer matches that of
a bridged or non-hostonly network.

Bridged Network Address: '10.0.0.0'
Host-only Network 'en0: Wi-Fi (Wireless)': '10.0.0.0'
```

A quick fix would be to update the `ip` field near the end of the Vangrantfile to a different block, such as `10.1.0.#{100 +i}`.

Reply:
Okay cool - I'll update the Vagrantfile to (hopefully) prevent most people from hitting this.